### PR TITLE
feat: add asana component

### PR DESCRIPTION
**Pull request by YCK1130**
Because

- We need Asana component to fulfill user story

This commit

- Implement most functions about goal, task, project, portfolio

### CODE REVIEW

File: application/asana/v0/config/definition.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "availableTasks": [
+    "TASK_ASANA_GOAL",
+    "TASK_ASANA_TASK",
+    "TASK_ASANA_PROJECT",
+    "TASK_ASANA_PORTFOLIO"
```
Comment (chuang8511) [inline application/asana/v0/config/definition.json:6]:
```suggestion
    "TASK_CRUD_GOAL",
    "TASK_CRUD__TASK",
    "TASK_CRUD__PROJECT",
    "TASK_CRUD__PORTFOLIO"
```
how about it?

File: application/asana/v0/config/setup.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "additionalProperties": false,
+  "properties": {
+    "token": {
+      "description": "Fill in your Asana Personal Access Token (PAT). You can generate one from <a href=\"https://app.asana.com/0/my-apps\">developer console<a/>",
```
Comment (chuang8511) [inline application/asana/v0/config/setup.json:6]:
```suggestion
  "additionalProperties": false,
  "properties": {
    "token": {
      "description": "Fill in your Asana Personal Access Token (PAT). You can generate one from [developer console](https://app.asana.com/0/my-apps)",
```

File: application/asana/v0/config/tasks.json [hunk 3]
```diff
@@ -0,0 +1,2673 @@
+{
+  "TASK_ASANA_GOAL": {
```
Comment (chuang8511) [inline application/asana/v0/config/tasks.json:2]:
```suggestion
  "TASK_CRUD_GOAL": {
```

File: application/asana/v0/config/tasks.json [hunk 4]
```diff
@@ -0,0 +1,2673 @@
...
+    "description": "Goal Related Task",
```
Comment (chuang8511) [inline application/asana/v0/config/tasks.json:3]:
```suggestion
    "description": "Create, Update, Delete, Read goal in Asana",
```

File: application/asana/v0/config/tasks.json [hunk 5]
```diff
@@ -0,0 +1,2673 @@
...
+          "title": "Task Notes in HTML format",
+          "instillFormat": "string",
+          "type": "string"
+        },
+        "projects": {
+          "description": "The projects of the task.",
+          "instillShortDescription": "The projects of the task.",
+          "instillUIOrder": 4,
+          "title": "Projects",
+          "instillFormat": "array",
```
Comment (chuang8511) [inline application/asana/v0/config/tasks.json:1306]:
```suggestion
          "instillFormat": "array:object",
```

File: application/asana/v0/config/tasks.json [hunk 6]
```diff
@@ -0,0 +1,2673 @@
...
+          "title": "Liked",
+          "instillFormat": "boolean",
+          "type": "boolean"
+        },
+        "likes": {
+          "description": "The users who likes the goal.",
+          "instillShortDescription": "The users who likes the goal.",
+          "instillUIOrder": 7,
+          "title": "Likes",
+          "instillFormat": "array",
```
Comment (chuang8511) [inline application/asana/v0/config/tasks.json:438]:
```suggestion
          "instillFormat": "array:object",
```

File: application/asana/v0/config/tasks.json [hunk 7]
```diff
@@ -0,0 +1,2673 @@
...
+          "title": "Liked",
+          "instillFormat": "boolean",
+          "type": "boolean"
+        },
+        "likes": {
+          "description": "The users who likes the task.",
+          "instillShortDescription": "The users who likes the task.",
+          "instillUIOrder": 7,
+          "title": "Likes",
+          "instillFormat": "array",
```
Comment (chuang8511) [inline application/asana/v0/config/tasks.json:1363]:
```suggestion
          "instillFormat": "array:object",
```

File: application/asana/v0/config/setup.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (chuang8511) [inline application/asana/v0/config/setup.json:1]: Do you miss `base-url`?

File: application/asana/v0/goal.go [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+	// req := c.Client.R().SetResult(&GoalTaskResp{}).SetBody(map[string]interface{}{
+	// 	"data": &UpdateGoalReq{
+	// 		Name:    input.Name,
+	// 		Notes:   input.Notes,
+	// 		DueOn:   input.DueOn,
+	// 		StartOn: input.StartOn,
+	// 		Liked:   input.Liked,
+	// 		Status:  input.Status,
+	// 	},
+	// })
```
Comment (chuang8511) [inline application/asana/v0/goal.go:127]:
```suggestion

```

**Review by chuang8511 — COMMENTED** [review]
Hi @YCK1130 
If you'd like to fix it, please let me know.
If you do not have time, please also let me know. I will take over it!

File: application/asana/v0/config/setup.json (hunk 8 above)
Comment (YCK1130) [reply application/asana/v0/config/setup.json:1]: "base-url" is left only for the mock server. I will update the client.go/newClient to use default baseURL

File: application/asana/v0/config/setup.json (hunk 2 above)
Comment (YCK1130) [reply application/asana/v0/config/setup.json:6]: I'd like to know why I should change the link to md format. It was not working in the previous console UI or the documentation. Is it fixed now?

Comment (chuang8511) [reply application/asana/v0/config/setup.json]: Originally, the IDE shows error
And, now, I found the reason..
<a/> should be </a>
image

**Review by chuang8511 — APPROVED** [review]
lgtm

**Comment by kuroxx** [comment]
@YCK1130 cc @chuang8511  
Could you point me to the Asana doc reference you used to build this integration 🙏

**Comment by chuang8511** [comment]
cc @YCK1130 
I just rebased it because I merged Amelia's branch.

**Review by chuang8511 — APPROVED** [review]
lgtm

**Comment by chuang8511** [comment]
@YCK1130   cc @kuroxx 
Thanks for your contribution!
Could you provide your recipe when you test this component?
So, Anni or I can use it as a reference to verify it and use it as the marketing content!

**Comment by YCK1130** [comment]
Hi @chuang8511 @kuroxx [Here](https://developers.asana.com/reference/createportfolio) is the Asana documentation. It's an API playground and you can test API or find required GID here.

The recipe that fulfills the requirements of testing component functions and marketing purposes is sent to the Slack channel. However, it only works after the patch PR #328 is merged.

**Pull request merged**